docs/migration.md

```diff
@@ -0,0 +1,193 @@
+  }
+```
+
+Would look like this:
+
+```yml
+groups:
+- name: example.rules
+  rules:
+  - record: job:request_duration_seconds:99percentile
+    expr: histogram_quantile(0.99, sum(rate(request_duration_seconds_bucket[1m]))
+      BY (le, job))
+  - alert: FrontendRequestLatency
+    expr: job:request_duration_seconds:99percentile{job="frontend"} > 0.1
+    for: 5m
+    annotations:
+      summary: High frontend request latency
+```
+
+To help with the change, the `promtool` tool has a mode to automate the rules conversion.  Given a `.rules` file, it will output a `.rules.yml` file in the
+new format. For example:
+
+```
+$ promtool update rules example.rules
+```
+
+## Storage
+
+The data format in Prometheus 2.0 has completely changed and is not backwards
+compatible with 1.8. To retain access to your historic monitoring data we recommend
+you run a non-scraping Prometheus 1.8.1 instance in parallel with your Prometheus 2.0
+instance, and have the new server read existing data from the old one via the
+remote write protocol.
+
+Your Prometheus 1.8 instance should be started with the following flags and an
+empty config file (`empty.yml`):
+
+```
+$ ./prometheus-1.8.1.linux-amd64/prometheus -web.listen-address ":9094" -config.file empty.yml
+```
+
+NOTE: **NOTE** If you used external labels in your Prometheus 2.0 config, they need to be
+preserved in your Prometheus 1.8 config.
+Prometheus 2.0 can then be started (on the same machine) with the following flags:
+
+```
+$ ./prometheus-2.0.0.linux-amd64/prometheus --config.file prometheus.yml
+```
+
+Where `prometheus.yml` contains the stanza:
+
+```
+remote_read:
+  - url: "http://localhost:9094/api/v1/read"
+```
+
+## PromQL
+
+The follow features have been removed from PromQL:
+
+- `drop_common_labels` function
+- `count_scalar` function
+- `keep_common` aggregation modifier
+
+See [issue #3060](https://github.com/prometheus/prometheus/issues/3060) for more
+details.
+
+## Miscellaneous
+
+### Prometheus non-root user
+
+The Prometheus Docker image is now built to [run Prometheus
+as a non-root user](https://github.com/prometheus/prometheus/pull/2859).  If you
+want the Prometheus UI/API to listen on a low port number (say, port 80), you'll
+need to override it.  For Kubernetes, you would use the following YAML:
+
+```yml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: security-context-demo-2
+spec:
+  securityContext:
+    runAsUser: 0
+...
+```
+
+See [https://kubernetes.io/docs/tasks/configure-pod-container/security-context/](Configure a Security Context for a Pod or Container)
+for more details.
+
+If you're using Docker, then the follow snippet would be used:
+
+```
+docker run -u root -p 80:80 prom/prometheus:v2.0.0-rc.2  --web.listen-address :80
+```
+
+## Prometheus lifecycle
+
+If you use the Prometheus `/-/reload` HTTP endpoint to [automatically reload your
+Prometheus config when it changes](https://www.weave.works/blog/prometheus-configmaps-continuous-deployment/),
```

Comment:
https://prometheus.io/docs/prometheus/latest/configuration/configuration/ would be a better link

Reply:
But the Weaveworks link explicitly talks about automating config reloads, which is the topic here?

Reply:
The weaveworks blog has its focus on k8 configmaps more than Prometheus config reloading. We have many users not on k8, and that blog also breaks with 2.0's security changes. I'd say we stick to our own simpler docs.

Reply:
Yeah I agree, I don't particularly want to link out to any third party blogs in the official docs.